### ISSUE TITLE
Stairs are treated as open walls for visibility calculations

### ISSUE THREAD

**Issue by julealgon** [diasurgical/devilutionX#1283]
**Important information**
Windows 10
Latest master https://github.com/diasurgical/devilutionX/commit/8425a23585063ff61481a8b805d8e32146be325d

**Describe**
When there are rooms behind stairs, the wall where the stair was placed is treated as open space, allowing the player to see into the room behind them.

**To Reproduce**
1. Find a map where there is a valid room right behind the stairs
2. Position yourself in front or close to the side of the stairs without entering it
3. Observe that the room behind the stairs become visible, and walls near the stairs become transparent

Save with the affected map layout
- [single_1.zip](https://github.com/diasurgical/devilutionX/files/6215004/single_1.zip)

**Expected behavior**
Stairs should not be "transparent" for visibility calculations. They should be treated as solid walls.

**Screenshots**
![HoleInStairsTransparency](https://user-images.githubusercontent.com/461579/112707529-ea273080-8e8a-11eb-9617-42e66ab22b5e.gif)

**Additional context**
I believe this should only happen with Cathedral and Crypt stairs, since caves/hive have no transparency and hell stairs are always positioned in the middle of the room.

**Comment by AJenbo** [comment]
interesting find

**Comment by galaxyhaxz** [comment]
I always noticed this, particularly in the PR demo old style dungeon code. It only happens on levels 3 and 4 where the staircase may have a room behind it. It is indeed bugged, checked out the visibility pathing. In two different spots, you have view behind the staircase and missiles can pass through as well.
![DIABLO_20210328_014221](https://user-images.githubusercontent.com/65138215/112739852-61b59800-8f3d-11eb-8bbe-22098b175876.png)

**Comment by galaxyhaxz** [comment]
A temporary way to fix this is to set the dPiece's with a red X to be solid and move the rigger from the blue dot to either of the green dots. The tiles directly behind the stair case on the left side are not solid as they are floor tiles, so you can't make them solid as it would mess up the whole floor. Unless you create a per-tile flag map like they did on the PSX.
![DIABLO_20210328_021244](https://user-images.githubusercontent.com/65138215/112740294-5cf2e300-8f41-11eb-8b02-00498e2f0d2f.png)

**Comment by AJenbo** [comment]
What's PSX?

**Comment by qndel** [comment]
> What's PSX?

https://letmegooglethat.com/?q=psx

**Comment by AJenbo** [comment]
wow pretty cool that Pakistan has started to work on the code

**Comment by galaxyhaxz** [comment]
Psx = PlayStation port

Derp :P

**Comment by AJenbo** [comment]
I hate all these pr naming schemes, why couldn't they just stick with normal numbers and call it PS10 instead so there wouldn't be this confusion.

**Comment by galaxyhaxz** [comment]
haha, who knows, maybe they should use Yiddish abbreviations instead ;)

**Comment by julealgon** [comment]
@AJenbo 
> What's PSX?
This is an actual Sony PSX: https://leganerd.com/wp-content/uploads/2013/02/psx01.jpg

I have no idea why people invented the PSX term to refer to PS1, but it is _probably_ incorrect considering actual PSX hardware exists and is something else entirely.

**Comment by AJenbo** [comment]
PSX was originally the codename for the PlayStation

**Comment by galaxyhaxz** [comment]
Yep, if you look on the diablo disc you will see the files named `DIABPSX` as the project codename.

**Comment by AJenbo** [comment]
Probably the dev tools also reflected this naming.

**Comment by julealgon** [comment]
> PSX was originally the codename for the PlayStation

Interesting. I guess I should've know this considering how much of a console player I consider myself 😆

**Comment by galaxyhaxz** [comment]
It's not really your fault either. Sony made it very confusing when they released the PSX, as it was a PS2 with more features it didn't warrant calling it the PS3. I think PS2X would have been better.

**Comment by AJenbo** [comment]
Yeah Sony, take a note from MS already will you. It goes `1`, `2`, `3`, `95`, `98`, `me`, `xp`, `vista`, `7`, `8`, `10`. Or if you want to be gaming specific `''`, `360`, `one`, `series`. Simple!
